code/modules/events/flutes.dm

```diff
@@ -0,0 +1,142 @@
+//
+//add ability to cure sanity instability with psychatrist?
+//add goals?
+//xoxeyos
+//massive help from Art Prime
+
+//---------
+//UTILITIES
+//---------
+/proc/list_intersection(var/list/A, var/list/B)
+	var/intersection[0]
+	for(var/item in A)
+		if(item in B)
+			intersection.Add(item)
+	return intersection
+
+//----------------------------
+//ROUND EVENT CONTROL "FLUTES"
+//----------------------------
+/datum/round_event_control/flutes
+	name = "Flutes"
+	typepath = /datum/round_event/flutes
+	max_occurrences = 1
+	weight = 1
+	min_players = 30
+
+//--------------------
+//ROUND EVENT "FLUTES"
+//--------------------
+//OBJECT DEF
+/datum/round_event/flutes
+	var/chosen_players[0]
+
+//ROUND START PROC
+//This gets executed at the start of the round
+/datum/round_event/flutes/start()
+	var/list/remaining_players = list_intersection(GLOB.player_list, GLOB.alive_mob_list)
+	var/number_of_players_to_select = round((remaining_players.len / 10), 1)
+	for(var/i; i<=(number_of_players_to_select == 0 ? 1 : number_of_players_to_select); i++)
+		var/player = remaining_players[rand(1, remaining_players.len)]
+		sound_intro(player)
+		pick_flute_scene(player)
+		remaining_players.Remove(player)
+		chosen_players.Add(player)
+
+//EVENT PROCS
+/datum/round_event/flutes/proc/sound_intro(mob/living/carbon/M)
+	if((M.client.prefs.toggles & SOUND_MIDI) && is_station_level(M.loc.z))
```

Comment:
I dont get why:
1) People with no ambience sound should get an informative sound disabled
2) People off of the z level should get an informative sound disabled